tests/vm_test.go

```diff
@@ -1921,6 +1922,91 @@ status:
 			}, 2*time.Minute, 1*time.Second).Should(Succeed())
 		})
 	})
+
+	Context("when node becomes unhealthy", Serial, func() {
+		var nodeName string
+
+		AfterEach(func() {
+			node, err := kubevirt.Client().CoreV1().Nodes().Get(context.Background(), nodeName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			node.Spec.Unschedulable = false
+			node, err = kubevirt.Client().CoreV1().Nodes().Update(context.Background(), node, metav1.UpdateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(node.Spec.Unschedulable).To(BeFalse())
+
+			libnode.Untaint(nodeName, "node.kubernetes.io/unschedulable")
+		})
+		It("[Serial] the VMs running in that node should be respawned", func() {
+			var pod *k8sv1.Pod
+			var wg sync.WaitGroup
+
+			By("Starting VM")
+
+			vmi := libvmi.NewAlpine()
+			vmi.Namespace = testsuite.GetTestNamespace(vmi)
+			vm := createRunningVM(virtClient, vmi)
+
+			vm = startVM(virtClient, vm)
+
+			vmi, err = virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, &k8smetav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Taint the node, it simulates that something went wrong in the cluster and workloads needs to be removed")
+
+			nodeName = vmi.Status.NodeName
+			libnode.Taint(nodeName, "node.kubernetes.io/unschedulable", k8sv1.TaintEffectNoExecute)
+
+			node, err := kubevirt.Client().CoreV1().Nodes().Get(context.Background(), nodeName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Disable scheduling in the node")
+
+			node.Spec.Unschedulable = true
+			node, err = kubevirt.Client().CoreV1().Nodes().Update(context.Background(), node, metav1.UpdateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("VMi should be NotReady")
+
+			Eventually(ThisVMI(vmi), 60*time.Second, 2*time.Second).Should(HaveConditionFalse(v1.VirtualMachineInstanceReady))
+
+			By("Pods should be in terminating state")
+
+			Eventually(func() bool {
+				pod, err = libvmi.GetPodByVirtualMachineInstance(vmi, testsuite.GetTestNamespace(vmi))
+				Expect(err).ToNot(HaveOccurred())
+				return pod.DeletionTimestamp != nil
+			}, 30*time.Second, 2*time.Second).Should(BeTrue())
+
+			vmiWatch, err := kubevirt.Client().VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Watch(context.Background(), metav1.ListOptions{})
```

Comment:
Move definition of vmiWatch to above where it is used? Inside the goroutine.

Reply:
Dropped.